### PR TITLE
Omit `...{:type}` annotations from search index

### LINKED ISSUE #2913: [v3] Extra {:foo} annotations not stripped from search text
When writing something like 
```
`<script></script>{:html}`
```
the `{:html}` appears as part of the search text when using the search box. It should be stripped out before the text is added to the search box.

### PR DESCRIPTION
**Pull request by siefkenj**
Fixes #2913 by omitting the `{:type}` annotations from the search index when they appear in inline code.

### CODE REVIEW

File: packages/nextra/src/server/remark-plugins/remark-structurize.ts [hunk 1]
```diff
@@ -56,10 +59,18 @@ export const remarkStructurize: Plugin<[Search], Root> = options => {
...
+      type === 'inlineCode' ||
+      type === 'tableCell'
     ) {
-      result += node.value
-      if (!skip) content += node.value
+      // Because of markdown extensions, inline code may have a `{:type}` suffix.
+      // Trim this off if it exists.
+      const value =
+        type === 'inlineCode'
+          ? node.value.replace(/\s*\{:\w+\}$/, '')
```
Comment (dimaMachina) [inline packages/nextra/src/server/remark-plugins/remark-structurize.ts:70]:
```suggestion
      // Inline code may have a `{:language}` suffix, trim this off if it exists.
      const value =
        type === 'inlineCode'
          ? node.value.replace(/\{:\w+\}$/, '')
```

i think `\s*` is useless here

**Comment by siefkenj** [comment]
@dimaMachina Are there any other changes needed on this PR?

**Comment by dimaMachina** [comment]
I am currently busy with other tasks, no worries I will include this change in next release when I will back to clean up Nextra for contributions

**Review by dimaMachina — APPROVED** [review]
LGTM, sorry for taking the time to merge this!

**Pull request merged**